Feature/adds authorization

raftx24 opened the pull request.

raftx24 committed:
adds `Authorizes`

raftx24 committed:
adds params to data-import table

raftx24 committed:
adds risky

raftx24 committed:
fixes

aocneanu commented on src/Exceptions/DataImport.php:
```diff
@@ -20,4 +20,9 @@ public static function fileNotReadable()
+
+    public static function unauthorized()
+    {
+        return new static(__("This action is unauthorized"));
```
You are not authorized to perform this import

aocneanu commented on src/Services/Importers/Chunk.php:
```diff
@@ -57,7 +59,8 @@ public function __construct(
     {
-        $this->auth();
+        $this->authenticate();
+        $this->authorize();
```
authorize should be before authenticate

aocneanu commented on src/Services/Importers/Chunk.php:
```diff
@@ -69,13 +72,21 @@ public function run(): void
+    private function authorize(): void
+    {
+        if ($this->importer instanceof Authorizes
+            && $this->importer->authorize($this->user, $this->params) === false) {
```
why not use `!` here?

raftx24 replied:
because I want to use laravel convention here
only `return false` is assumed as `unauthorized`

aocneanu commented on src/Contracts/Authorizes.php:
```diff
@@ -0,0 +1,11 @@
+
+interface Authorizes extends Authenticates
+{
+    public function authorize(User $user, Obj $params);
```
rename to `authorizes`

raftx24 committed:
adds upgrade
moves params after type

raftx24 committed:
address reviews
- rename authorize method authorizes
- reorder method authentication & authorization
- improve exception message

raftx24 committed:
Merge pull request #43 from laravel-enso/fixes/stylci

Fixes/stylci

aocneanu committed:
Apply fixes from StyleCI

aocneanu commented on src/Contracts/Authorizes.php:
```diff
@@ -0,0 +1,11 @@
+
+interface Authorizes extends Authenticates
+{
+    public function authorizes(User $user, Obj $params);
```
:bool

raftx24 committed:
adds return type to authorize methode

raftx24 committed:
Merge remote-tracking branch 'origin/feature/addsAuthorization' into feature/addsAuthorization
